Ftrack: Support multiple reviews

iLLiCiTiT opened the pull request:
## Changes
- rewritten plugin to easier find out thumbnail, review and others representations
- first review representation is published as before
    - all following review representations will be published to different ftrack asset
    - ftrack asset name in that case is made up with concatenation of subset name and representation name

OpenPype 3 version of https://github.com/pypeclub/OpenPype/pull/1374

mkolar commented:
Task linked: [OP-1969 Support multiple Ftrack reviews](https://app.clickup.com/t/6658547/OP-1969)

mkolar reviewed (changes requested):
The second reviewable doesn't get a thumbnail on ftrack so it looks like it's not working even though the review is there. 
Secondly, I think we should Concatenate the names with `_` it will be cleaner in the ftrack views

iLLiCiTiT commented:
> I think we should Concatenate the names with _ it will be cleaner in the ftrack views

Fine for me

> The second reviewable doesn't get a thumbnail on ftrack so it looks like it's not working even though the review is there.

I don't know how to pair thumbnail to review (but we definetely should have ability to do so).

mkolar commented:
>I don't know how to pair thumbnail to review (but we definetely should have ability to do so).

well, you need to actually define it as a component on the version. We're already doing it for all reviewable media we upload.

iLLiCiTiT commented:
I've modified it
- fixed bug for "other components" where component path was not set
- not first reviews have ftrack asset name created with template `{subset name}_{representation name}`
- not first reviews will duplicate thumbnail for the new asset
